library/webpack-plugins/common-chunks-plugin.js

```diff
@@ -0,0 +1,376 @@
+    },
+    { exclusive: [], shared: [] }
+  )
+}
+
+function determineShareCounts(newChunks, affectedChunks) {
+  return Array.from(affectedChunks.values()).reduce(
+    (result, newChunks) => {
+      newChunks.forEach(newChunk => {
+        const previousShareCount = result.get(newChunk) || 0
+        const newShareCount = newChunks.length
+
+        if (newShareCount > previousShareCount) result.set(newChunk, newShareCount)
+      })
+      return result
+    },
+    new Map()
+  )
+}
+
+/*
+  This function is not strictly necessary but allows us to, more easily catch edge cases
+  and changes to the webpack internals.
+
+  With this function we check if each original chunk that was affected by this module
+  has exactly 1 runtime in it's 'chunk-chain'
+*/
+function dubbelCheckRuntimes(newChunks) {
+  const affectedChunks = determineAffectedChunks(newChunks)
+  /*
+    affectedChunks = Map(
+      [chunk]: [newChunk1, ..., newChunkN]
+    )
+  */
+  Array.from(affectedChunks.entries()).forEach(
+    ([chunk, newChunks]) => {
+
+      if (chunk.hasRuntime()) error(
+        `Optimized chunk (${chunk.name}) has a runtime while it uses ${newChunks.length} shared chunks`
+      )
+
+      const newChunksWithRuntime = newChunks.filter(newChunk => newChunk.hasRuntime())
+      if (newChunksWithRuntime > 1) error(
+        `More than one shared chunk (used by a single chunk), ${newChunksWithRuntime.length} to be exact,\n` +
+        `has a runtime:\n` +
+        `${newChunksWithRuntime.map(c => c.name).join(', ')}`
+      )
+
+      if (newChunksWithRuntime === 0) {
+        newChunks.reduce(
+          (previousParent, newChunk) => {
+            const parent = newChunk.parent
+
+            if (!parent.hasRuntime()) error(
+              `Shared chunk (${newChunk.name}) does not have a runtime and it's parent (${parent.name}) does not have\n` +
+              `one either.`
+            )
+
+            if (previousParent && previousParent !== parent) error(
+              `Shared chunk (${newChunk.name}) does not have a runtime and it's parent (${parent.name}) is different\n` +
+              `from the parent of the previous shared chunk (${previousParent.name}). This means the chunk (${chunk.name})\n` +
+              `has more than one runtime.`
+            )
+
+            return parent
+          },
+          null
+        )
+      }
+    }
+  )
+
+  function error(message) {
+    throw new Error(
+      `You have found an implementation problem in the CommonChunksPlugin of @kaliber/build.\n` +
+      `This error means one of two things:\n` +
+      `- Webpack has changed it's internals and we need to check the changes to make this plugin\n` +
+      `  work again.\n` +
+      `- You have found a situation that we did not foresee when writing this plugin\n` +
+      `\n` +
+      `* In both cases it would be awesome if you were to create an issue. *\n` +
+      `\n` +
+      `Extra information for the developers:\n` +
```

Comment:
etc

Reply:
Haha, generally I get the point when you mention it once, but still, it's nice you point it out multiple times. I also tend to forget stuff... Haha